node-rfc/run_rfc_task.js

```diff
@@ -0,0 +1,209 @@
+                            grunt.log.errorlns("Error invoking", functionModule, err);
+                            return reject();
+                        }
+                        client.close();
+                        grunt.log.writeln("Messages:", res.EV_LOG_MESSAGES);
+                        return resolve(res);
+                    });
+            });
+        });
+    };
+
+
+    grunt.registerTask("createTransportRequest", "Creates an ABAP Transport Request", function() {
+        grunt.log.writeln("Creating Transport Request");
+        var importParameters = {
+            AUTHOR: this.options().author,
+            TEXT: this.options().description
+        };
+        var done = this.async();
+        rfcConnect("BAPI_CTREQUEST_CREATE", importParameters, this)
+            .then(
+            function(returnValue) {
+                if (returnValue.EV_SUCCESS == "E" || returnValue.EV_SUCCESS == "W") {
+                    grunt.log.errorlns("Error invoking BAPI_CTREQUEST_CREATE.");
+                    grunt.log.errorlns("Message Id:", returnValue.EV_MSG_ID);
+                    grunt.log.errorlns("Message No:", returnValue.EV_MSG_NO);
+                    grunt.log.errorlns("Messages:", returnValue.EV_LOG_MESSAGES);
+                    done(false);
+                    return;
+                }
+                if (returnValue.REQUESTID == "") {
+                    grunt.log.errorlns("Error invoking BAPI_CTREQUEST_CREATE.");
+                    grunt.log.errorlns("Transport request could not be created.");
+                    grunt.log.errorlns(returnValue.RETURN.MESSAGE);
+                    done(false);
+                    return;
+                }
+                grunt.log.writeln("Transport request", returnValue.REQUESTID, "created.");
+                if (fs.existsSync(targetDir) === false) {
+                    fs.mkdirSync(targetDir);
+                }
+                fs.writeFile(ctsDataFile,
+                    JSON.stringify(
+                        { REQUESTID: returnValue.REQUESTID }
+                    ),
+                    function(err) {
+                        if (err) {
+                            grunt.log.errorlns("Error Creating file:", err);
+                            done(false);
+                            return;
+                        }
+                        grunt.log.writeln("Created file:", ctsDataFile);
+                        done();
+                    }
+                )
+            },
+            function() {
+                done(false);
+            });
+    });
+
+    grunt.registerTask("uploadToABAP", "Uploads the application to the ABAP System", function(transportRequest) {
+        grunt.log.writeln("Uploading to ABAP");
+        if (!transportRequest) {
+            if (!fs.existsSync(ctsDataFile)) {
+                grunt.log.errorlns("No Transport request specified. Pass one explicitly or run createTransportRequest first.");
```

Comment:
What is promissed in the message here will not be fullfilled. Since we launch a different container each time it will not be possible to create a transport request and to refer to the id provided in a file in a second call.

Reply:
In the README.md there is a description that the /var/sap/data should be mounted from outside to keep intermediate data.